packages/default-nav/ml/default_navigation.ts

```diff
@@ -127,12 +135,21 @@ export const defaultNavigation: MlNodeDefinition = {
       children: [
         {
           link: 'ml:logRateAnalysis',
+          getIsActive: ({ pathNameSerialized, prepend }) => {
```

Comment:
Why is `getIsActive` needed for these 4 links, but not any of the others?
Commenting these out, I can't see any change in behaviour?

Reply:
ml:logRateAnalysis leads to `/log_rate_analysis_index_select`, but when you select a data view, you redirect to `/log_rate_analysis`
we need this override so that navigation tree considered `link: 'ml:logRateAnalysis'` active for both `/log_rate_analysis_index_select` and `/log_rate_analysis`, otherwise breadcrumb and active highlight in side nav doesn't work:

![Screenshot 2023-10-24 at 13 19 03](https://github.com/elastic/kibana/assets/7784120/14f2e56d-89b8-4ef2-ba43-d32b6f3716d0)

You probably didn't see the difference because [packages/default-nav/ml/default_navigation.ts](https://github.com/elastic/kibana/pull/169513/files/6fb9882edac163b222d2f07b2d4a570d8db1aba1#diff-d93b4c75f0535e7c71aeb02c1de52d02ff95120c0e47722551d78c0dbd8eac0f) is used in security project, but observability built a nav tree [here](x-pack/plugins/serverless_observability/public/components/side_navigation/index.tsx)

Reply:
Ah ok, thanks for the explanation.
`ml:indexDataVisualizer` is similar, it also points to an "index select" page before redirecting to the main page.

Reply:
Thanks! adjusted